### ISSUE TITLE
UrlPath env option

### ISSUE THREAD

**Issue by adrianipopescu** [riffsphereha/newznabarr#5]
It would be amazing to have the ability to host via reverse proxy in a subfolder, but I can also understand this being a fully-internal docker network project, with the ui being exposed just for initial debugging.

**Comment by riffsphereha** [comment]
Pretty sure no paths are hardcoded.
Sure, the /api is, but at no point in the code do I reference ip:port/api I believe, so I don't see why url/subfolder/api wouldn't work?

**Comment by adrianipopescu** [comment]
In other languages I've usually ran into issues with straight / or /api, with the info being sent to the browser directly when generating links or as an inverse, when receiving extra parameters with the app not knowing its "core dir", it would add extra items to the url, via the parser.

ex: if I host at /nwznabr/ and try to hit the /api route, in C# in older setups it would just look at the request url and parse that, and not match /nwznabr/api to /api, since extra tokens.

ex2: when generating links without knowing the base url, it would always ignore the base url path, so I'd try to create a url to /api from serverside / clientside js it would generate (correctly) based on the current url with the / acting like in linux -> root, meaning if I'd want a @Html.ActionLink (or whatever it was called, it's been years) to a route, it wouldn't take into account any additional url context since it had no reference on "where to start" or "what the / root is"

tl;dr edition: I'll give it a shot and report back

**Comment by adrianipopescu** [comment]
Tried it this morning, it instantly gave a 404, using the same traefik config I use for other services. I'll keep tinkering with it.
